test/python/algorithms/test_vqe.py

```diff
@@ -586,25 +586,25 @@ def test_aux_operator_std_dev_pauli(self):
         self.assertEqual(len(result.aux_operator_eigenvalues), 2)
         # expectation values
         self.assertAlmostEqual(result.aux_operator_eigenvalues[0][0], 2.0, places=6)
-        self.assertAlmostEqual(result.aux_operator_eigenvalues[1][0], 0.5784419552370315, places=6)
+        self.assertAlmostEqual(result.aux_operator_eigenvalues[1][0], 0.6796875, places=6)
```

Comment:
This is what I get on **main** when I use expectation factory with a statevector instead of qasm for this test:
```
F
======================================================================
FAIL: test_aux_operator_std_dev_pauli (__main__.TestVQE)
__main__.TestVQE.test_aux_operator_std_dev_pauli
----------------------------------------------------------------------
testtools.testresult.real._StringException: Traceback (most recent call last):
  File "qiskit-terra/test/python/algorithms/test_vqe.py", line 594, in test_aux_operator_std_dev_pauli
    self.assertAlmostEqual(result.aux_operator_eigenvalues[1][0], 0.5784419552370315, places=6)
  File "/unittest/case.py", line 906, in assertAlmostEqual
    raise self.failureException(msg)
AssertionError: 0.6698863565455352 != 0.5784419552370315 within 6 places (0.09144440130850373 difference)
```
The same code with an expectation factory and qasm passes on main. It shows that we have a significant discrepancy on main in the original VQE code between the statevector and qasm:  `0.6698863565455352 != 0.5784419552370315` .

Whereas after our change here, both backends give the result close to `0.6698863565455352` . To be precise: `0.6796875` for qasm and `0.6698863565455352` for the statevector. So the latter is exactly the same as on main after swapping qasm to statevector in this test.

It gives a good reason to assume that our version here is correct and by the way we fix some bug in the original VQE code.